### ISSUE TITLE
Clean up ES task/app registration

### ISSUE THREAD

**Issue by jphickey** [nasa/cFE#1215]
**Is your feature request related to a problem? Please describe.**
In nasa/osal#853 it is proposed that `OS_TaskRegister()` be finally deprecated/removed.

But this function is invoked by two places in CFE ES that follow a similar pattern: `CFE_ES_RegisterApp()` and `CFE_ES_RegisterChildTask()`.

**Describe the solution you'd like**
At a minimum, calls to `OS_TaskRegister()` must be removed to allow the function to be deprecated.

Furthermore there is already a task startup wrapper in ES that can be used to call/handle setting up environment (`CFE_PSP_SetDefaultExceptionEnvironment()`) meaning that these two functions themselves can also be deprecated - basically using the same design pattern as OSAL uses such that we don't need to burden apps with calling this extra function.  This design is simpler and less error prone.

**Additional context**
See nasa/osal#853

**Requester Info**
Joseph Hickey, Vantage Systems, Inc.

**Comment by jphickey** [comment]
Ping @skliper , @ejtimmon , @acudmore , @jwilmot ....

This is the only ref to OS_TaskRegister() that I found but interestingly we have basically the same situation in CFE ES - a couple functions that have effectively become no-ops - or at least unnecessary - but remain in the API for compatibility reasons.

However unlike OS_TaskRegister it looks like these ARE invoked by many CFS apps - so there are definite backward compatibility issues to removing/deprecating it.

This makes me think we should just leave it alone - is it that bad to have a couple no-op functions laying around for backward compatibility?

**Comment by skliper** [comment]
I find it more confusing to have an API that's half utilized, possibly partly documented, doesn't actually do anything, not really clear what the plan is for the future (will we leave it forever?), etc.  I'd rather a quick/hard break than the uncertainty the noops cause.  Unless we can fully utilize/document and very clearly state to keep these calls in (for future development or whatever), as it it's a bug if they are missing.

Basically half way is worse (in my view) than hard yes or hard no.

**Comment by skliper** [comment]
CCB - if we are going to do it, do it now.  No hard objects, lets remove.

**Comment by skliper** [comment]
Also noted on community-email list the comment below is misleading.  As long as we are removing then no additional action needed.  If for some reason this stays, should be updated.

https://github.com/nasa/cFE/blob/fa10af72f95a3a36b65352f0688498449487265f/modules/es/fsw/src/cfe_es_api.c#L663

ping @johnphamngc
